packages/guides/src/Compiler/NodeTransformers/MenuNodeTransformers/GlobMenuEntryNodeTransformer.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers\MenuNodeTransformers;
+
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Nodes\Menu\GlobMenuEntryNode;
+use phpDocumentor\Guides\Nodes\Menu\InternalMenuEntryNode;
+use phpDocumentor\Guides\Nodes\Menu\MenuEntryNode;
+use phpDocumentor\Guides\Nodes\Menu\MenuNode;
+use phpDocumentor\Guides\Nodes\Menu\NavMenuNode;
+use phpDocumentor\Guides\Nodes\Menu\TocNode;
+use phpDocumentor\Guides\Nodes\Node;
+
+use function array_pop;
+use function assert;
+use function explode;
+use function implode;
+use function in_array;
+use function preg_match;
+use function str_replace;
+use function str_starts_with;
+
+final class GlobMenuEntryNodeTransformer extends AbstractMenuEntryNodeTransformer
+{
+    // Setting a default level prevents PHP errors in case of circular references
+    private const DEFAULT_MAX_LEVELS = 10;
+
+    /** @return list<MenuEntryNode> */
+    protected function handleMenuEntry(MenuNode $currentMenu, MenuEntryNode $node, CompilerContext $compilerContext): array
+    {
+        assert($node instanceof GlobMenuEntryNode);
+        $maxDepth = (int) $currentMenu->getOption('maxdepth', self::DEFAULT_MAX_LEVELS);
+        $documentEntries = $compilerContext->getProjectNode()->getAllDocumentEntries();
+        $currentPath = $compilerContext->getDocumentNode()->getFilePath();
+        $globExclude = explode(',', $currentMenu->getOption('globExclude') . '');
+        $menuEntries = [];
+        foreach ($documentEntries as $documentEntry) {
+            if (
+                !self::isEqualAbsolutePath($documentEntry->getFile(), $node, $currentPath, $globExclude)
+                && !self::isEqualRelativePath($documentEntry->getFile(), $node, $currentPath, $globExclude)
+            ) {
+                continue;
+            }
```

Comment:
As the isEqualAbsolute and isEqualRelative path are both containing equal logic I would merge them into a single method. 

```suggestion
            if (!self::matches($documentEntry->getFile(), $node, $currentPath, $globExclude)) {
                continue;
            }
```

Reply:
Moved it